narwhal/primary/src/core.rs

```diff
@@ -448,6 +451,11 @@ impl Core {
         // Store the certificate.
         self.certificate_store.write(certificate.clone())?;
 
+        self.highest_processed_round = self.highest_processed_round.max(certificate.round());
+        self.metrics
+            .highest_processed_round
+            .with_label_values(&[&certificate.epoch().to_string()])
+            .set(self.highest_processed_round as i64);
         let certificate_source = if self.name.eq(&certificate.header.author) {
```

Comment:
Would it be useful to add the `certificate_source` as extra bit of information to the metric? It might be interesting to see whether we are listening for higher rounds from other nodes rather than from our own proposer.

Reply:
Sounds good, although I expect them to be very close.